pkg/gitutil/git.go

```diff
@@ -51,8 +51,11 @@ func update(destinationPath string) error {
 		return errors.Wrapf(err, "fetch index at %q failed", destinationPath)
 	}
 
-	err := exec(destinationPath, "reset", "--hard", "@{upstream}")
-	return errors.Wrapf(err, "reset index at %q failed", destinationPath)
+	if err := exec(destinationPath, "reset", "--hard", "@{upstream}"); err != nil {
```

Comment:
probably need to update godoc, and possibly the method name, right? 
this method now does certain things.

Also does this respect `--no-update-index` flags?

Reply:
put it under another function for clarity.

flag is already taken care in the call path.